linker/linker_test.go

```diff
@@ -1638,6 +1676,32 @@ func TestLinkerValidation(t *testing.T) {
 					t.Errorf("expecting validation error %q; instead got: %q", tc.expectedErr, err)
 				}
 			}
+
+			// parse with protoc
+			var passProtoc bool
+			if tc.inputOrder != nil {
```

Comment:
Aside: it doesn't make much difference in this particular case, but as a general practice, it is better to use `len(tc.inputOrder) == 0`. It is rarely useful to distinguish between nil and empty slices. (And not all empty slices are nil, though they will be in this particular test.)

So if you move this condition into `testByProtoc` (per other comment), maybe also tweak the condition to check length instead of looking for nil.

Reply:
Changed this into checking length instead of checking for nil. I remember seeing this in the go code standard but still keeping forgetting which one is correct.